k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -105,270 +103,277 @@ class SingleNsNamer(
   }
 }
 
+object SingleNsNamer {
+  protected val PrefixLen = 2
+}
+
 abstract class EndpointsNamer(
   idPrefix: Path,
-  mkApi: String => NsApi,
-  backoff: Stream[Duration] = Backoff.exponentialJittered(10.milliseconds, 10.seconds)
-)(implicit timer: Timer = DefaultTimer) extends EnumeratingNamer {
+  mkApi: String => v1.NsApi,
+  labelName: Option[String] = None,
+  backoff: Stream[Duration] = EndpointsNamer.DefaultBackoff
+)(implicit timer: Timer = DefaultTimer)
+  extends Namer with Stabilize {
 
   import EndpointsNamer._
+  protected[this] val variablePrefixLength: Int
```

Comment:
that's fair....would still like to factor out some stuff from `lookup` but it is probably less possible than I originally thought.